File: Python/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) == 0:
            return 0
        ans, mi = 0, prices[0]
        for price in prices:
            if price < mi:
                mi = price
            ans = max(ans, price - mi)
        return ans

File: C++/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int len = prices.size();
        int ans = 0;
        if(len==0) return 0;
        int mi = prices[0];
        for(int i=0;i<len;i++)
        {
            if(prices[i]<mi)
                mi = prices[i];
            ans = max(ans,prices[i]-mi);
        }
        return ans;
    }
};